File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DiagnosesRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface DiagnosesRow {
    DiagnosisId?: number;
    Code?: string;
    Name?: string;
    IsActive?: boolean;
}

export abstract class DiagnosesRow {
    static readonly idProperty = 'DiagnosisId';
    static readonly nameProperty = 'Code';
    static readonly localTextPrefix = 'Tanimlamalar.Diagnoses';
    static readonly deletePermission = 'Tanimlamalar.Diagnoses.General';
    static readonly insertPermission = 'Tanimlamalar.Diagnoses.General';
    static readonly readPermission = 'Tanimlamalar.Diagnoses.General';
    static readonly updatePermission = 'Tanimlamalar.Diagnoses.General';

    static readonly Fields = fieldsProxy<DiagnosesRow>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Diagnoses/DiagnosesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar;

[ConnectionKey("Default"), Module("Tanimlamalar"), TableName("Diagnoses")]
[DisplayName("Diagnoses"), InstanceName("Diagnoses")]
[ReadPermission("Tanimlamalar.Diagnoses.General")]
[ModifyPermission("Tanimlamalar.Diagnoses.General")]
[ServiceLookupPermission("Tanimlamalar.Diagnoses.General")]
public sealed class DiagnosesRow : Row<DiagnosesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Diagnosis Id"), Identity, IdProperty]
    public int? DiagnosisId { get => fields.DiagnosisId[this]; set => fields.DiagnosisId[this] = value; }

    [DisplayName("Code"), Size(200), QuickSearch, NameProperty]
    public string Code { get => fields.Code[this]; set => fields.Code[this] = value; }

    [DisplayName("Name"), Size(200)]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Is Active"), NotNull]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field DiagnosisId;
        public StringField Code;
        public StringField Name;
        public BooleanField IsActive;

    }
}